src/App.tsx

```diff
@@ -15,33 +16,68 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [isStarted, setStart] = useState(true);
+  const [sortType, setSort] = useState('');
+  const [initialOrder, setOrder] = useState(false);
+
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((good: any, nextGood: any) => {
+    switch (sortType) {
+      case 'alphabetically':
+        return good.localeCompare(nextGood);
+
+      case 'by length':
+        return good.length - nextGood.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return (
+    <>
+      <div className="App">
+        {isStarted && (
+          <button
+            type="button"
+            onClick={() => setStart(false)}
+          >
+            Start
+          </button>
+        )}
+
+        {!isStarted && (
+          <>
+            <button type="button" onClick={() => setSort('alphabetically')}>
+              Sort alphabetically
+            </button>
+
+            <button type="button" onClick={() => setSort('by length')}>
+              Sort by length
+            </button>
+
+            <button type="button" onClick={() => setReverse(true)}>
+              Reverse
+            </button>
+
+            <button type="button" onClick={() => setOrder(true)}>
+              Reset
+            </button>
+
+            {/* Can't sort or reverse the list items after I pressed reset button. Maybe it should be this way if no - let me know please */}
+            {!initialOrder
+              ? <GoodsList goods={visibleGoods} />
+              : <GoodsList goods={goodsFromServer} />}
```

Comment:
nope it shouldn't be this way, need to fix this

Reply:
Share the idea

Reply:
just create handlers and pass them to buttons, and when those button are clicked – perform the action (i.e. sort, reset, show – and so on)